### PR TITLE
feat: Adds update-libs workflow

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

Adds update-libs workflow to create PRs automaticaclly to update charm libraries.
The workflow will run every 12 hours.

This PR should not be merged before[ IS ticket ](https://portal.admin.canonical.com/C160234/)is resolved and `TELCO_GITHUB_BOT_TOKEN` secret is added.

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 import logging
 from typing import Dict, List
 
-from charms.tls_certificates_interface.v2.tls_certificates import (  # type: ignore[import]
+from charms.tls_certificates_interface.v2.tls_certificates import (  # type: ignore[import-not-found]  # noqa: E501
```
Comment (gruyaume) [inline src/charm.py:15]: You may want to rebase as this was fixed in PR #71

**Review by gruyaume — CHANGES_REQUESTED** [review]
Looks good besides rebase comment

File: .github/workflows/update-libs.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        run: |
+          sudo snap install charmcraft --classic --channel latest/stable
+          charmcraft fetch-lib
+        env:
+          CHARMCRAFT_AUTH: "${{ secrets.CHARMCRAFT_AUTH }}"
+
+      - name: Create a PR for local changes
+        uses: peter-evans/create-pull-request@v4.2.3 # what is this action?
+        with:
+          token: ${{ secrets.TELCO_GITHUB_BOT_TOKEN }} # what is this token? How to create it?
```
Comment (DanielArndt) [inline .github/workflows/update-libs.yaml:31]:
```suggestion
        uses: peter-evans/create-pull-request@v4.2.3
        with:
          token: ${{ secrets.TELCO_GITHUB_BOT_TOKEN }}
```

**Review by DanielArndt — CHANGES_REQUESTED** [review]

**Review by gruyaume — APPROVED** [review]

**Review by DanielArndt — APPROVED** [review]

File: .github/workflows/update-libs.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          mkdir -p ~/.ssh
+          echo "${{ secrets.BOT_PRIVATE_SIGNING_KEY }}" > ~/.ssh/id_rsa
+          chmod 600 ~/.ssh/id_rsa
+          echo "${{ secrets.BOT_PUBLIC_SIGNING_KEY }}" > ~/.ssh/id_rsa.pub
+          chmod 644 ~/.ssh/id_rsa.pub
+          git config --global user.email "telco-engineers@lists.canonical.com"
+          git config --global user.name "telcobot"
+          git config --global user.signingkey ~/.ssh/id_rsa.pub
+          git config --global commit.gpgsign true
+          git config --global gpg.format ssh
```
Comment (DanielArndt) [inline .github/workflows/update-libs.yaml:32]: Well... now I feel silly for not noticing this feature for the past two years...

File: .github/workflows/update-libs.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        env:
+          CHARMCRAFT_AUTH: "${{ secrets.CHARMCRAFT_AUTH }}"
+
+      - name: Create a PR for local changes
+        uses: peter-evans/create-pull-request@v4.2.3
+        with:
+          token: ${{ secrets.TELCO_GITHUB_BOT_TOKEN }}
+          commit-message: "chore: update charm libraries"
+          committer: "Github Actions <github-actions@github.com>"
+          author: "Github Actions <github-actions@github.com>"
```
Comment (DanielArndt) [inline .github/workflows/update-libs.yaml:47]: It seems odd to me that these don't match the `git config` settings above

File: .github/workflows/update-libs.yaml (hunk 3 above)
Comment (saltiyazan) [reply .github/workflows/update-libs.yaml:32]: :D

File: .github/workflows/update-libs.yaml (hunk 4 above)
Comment (DanielArndt) [reply .github/workflows/update-libs.yaml]: oops, looks like you only fixed the e-mail, not the name.

Either way, I'm not super concerned. Up to you if you think it's worth changing.

**Review by gruyaume — APPROVED** [review]

**Pull request merged**